point_pattern.py

```diff
@@ -0,0 +1,163 @@
+
+from point import Point
+import analytics
+import utils
+import random
+import numpy as np
+import scipy.spatial as ss
+import pysal as ps
+
+class PointPattern(object):
+    def __init__(self):
+        self.points = []
+
+    def add_point(self, point):
+        self.points.append(point)
+
+    def remove_point(self, index):
+        try:
+            del(self.points[index])
+        except:
+            pass
+
+    def average_nearest_neighbor_distance(self, mark=None):
+        return analytics.average_nearest_neighbor_distance(self.points, mark)
+
+
+    def average_nearest_neighbor_distance_kdtree(self, mark=None):
+        pointsWithMark = list()
+        pointsWithoutMark = list()
+
+        for p in self.points:
+            if mark != None:
+                if p.mark == mark:
+                    pointsWithMark.append(p.getPoint())
+            else:
+                pointsWithoutMark.append(p.getPoint())
+
```

Comment:
Here, if you wanted, you could utilize a single list.  If mark is None, the list is equal to all the points.  If mark is not None, then subset and append the same list.